client/src/main/java/com/arcbees/beeshop/client/application/product/ProductView.java

```diff
@@ -120,14 +128,20 @@
         this.brandPicker = brandPicker;
         this.appMessages = appMessages;
         this.productMessages = productMessages;
-        sizeMap = new HashMap<>();
+        sizeListItemAssociation = new HashMap<>();
+        sizeAnchorAssociation = new HashMap<>();
 
         initWidget(uiBinder.createAndBindUi(this));
 
-        sizeMap.put(Size.SMALL, smallAnchor);
-        sizeMap.put(Size.MEDIUM, mediumAnchor);
-        sizeMap.put(Size.LARGE, largeAnchor);
-        sizeMap.put(Size.XLARGE, xlargeAnchor);
+        sizeListItemAssociation.put(Size.SMALL, smallSizeListItem);
+        sizeListItemAssociation.put(Size.MEDIUM, mediumSizeListItem);
+        sizeListItemAssociation.put(Size.LARGE, largeSizeListItem);
+        sizeListItemAssociation.put(Size.XLARGE, xlargeSizeListItem);
```

Comment:
I don't think you can escape from the "anchors" map, but you could use a bit of gquery magic to remove the list items. It would look something like this:
In ui.xml:
```
<ui:import field="com.arcbees.beeshop.common.dto.Size.*"/>
...
<ul ui:field="sizes" class="{page.style.product_size}">
    <li data-size="{SMALL.getValue}">
```
And in the view:
```
$("li", sizes).removeClass(page.style.active());
$("li[data-size='" + productDto.getSize.getValue() + "'"], sizes).addClass(page.style.active());
```

I'm not sure of the exact gquery syntax, but should be enough to get the idea.